.circleci/config.yml

```diff
@@ -1,79 +1,115 @@
 version: 2.1
 
-commands:
-  install-packaged-autonormalize:
-    description: "install autonormalize from source distribution"
-    steps:
-      - run : |
-          source test_env/bin/activate
-          python setup.py sdist
-          AUTONORMALIZE_VERSION=$(python setup.py --version)
-          tar -zxvf "dist/autonormalize-${AUTONORMALIZE_VERSION}.tar.gz"
-          pip install -e "autonormalize-${AUTONORMALIZE_VERSION}/"
-          pip install -r "autonormalize-${AUTONORMALIZE_VERSION}/test-requirements.txt"
-  run-packaged-tests:
-    description: "run unit tests on packaged testing files"
-    steps:
-      - run: |
-          source test_env/bin/activate
-          cd "autonormalize-$(python setup.py --version)/"
-          pytest
-
-jobs:
-  "python-27":
-    working_directory: ~/auto-norm
+executors:
+  python:
+    parameters:
+      image_tag:
+        type: string
+        default: "python:3.7"
     docker:
-      - image: circleci/python:2.7
-    steps:
-      - checkout
-      - run: virtualenv test_env && virtualenv dev_env
-      - install-packaged-autonormalize
-      - run: source dev_env/bin/activate && make installdeps
-      - run: source dev_env/bin/activate && make lint
-      - run-packaged-tests
+      - image: circleci/<< parameters.image_tag >>
 
-  "python-35":
-    working_directory: ~/auto-norm
-    docker:
-      - image: circleci/python:3.5
+commands:
+  installation:
     steps:
       - checkout
-      - run: virtualenv test_env && virtualenv dev_env
-      - install-packaged-autonormalize
-      - run: source dev_env/bin/activate && make installdeps
-      - run: source dev_env/bin/activate && make lint
-      - run-packaged-tests
+      - run:
+          name: "Install Package in Virtual Environment"
+          command: |
+            virtualenv env -q
+            source env/bin/activate
+            pip install . --progress-bar off
```

Comment:
the benefit of making a packaged version to install from is that it can test for missing files in the uploaded package

it adds a few steps to the install process but avoids the headache of releasing a version with missing files

Reply:
will refactor to base on a packaged version

Reply:
refactored to use packaged version